gapic-generator-cloud/templates/cloud/gem/readme.erb

```diff
@@ -93,3 +101,45 @@ in security maintenance, and not end of life. Currently, this means Ruby 2.4
 and later. Older versions of Ruby _may_ still work, but are unsupported and not
 recommended. See https://www.ruby-lang.org/en/downloads/branches/ for details
 about the Ruby support schedule.
+<%- if gem.wrapper_name -%>
+
+## Which client should I use?
+
+Most modern Ruby client libraries for Google APIs come in two flavors:
+lower-level _versioned clients_ and higher-level _main clients_. As a TL;DR,
+for _most_ cases, you should install the main client.
+
+A _versioned client_ has a name such as `<%= gem.name %>`,
+and provides a basic set of client classes for a _single version_ of a specific
+service. Some services publish multiple versions of their API, with potentially
+different interfaces including differences in field names, types, or method
+calls. For such services, there may be a separate versioned client library for
+each service version. Most versioned clients are created and maintained by a
+code generator, based on the service's published interface descriptions.
+
+The _main client_ for a service has a name such as `<%= gem.wrapper_name %>`.
+There will be only one main client for any given service, even a service with
+multiple versions. For most services, the main client does not directly include
+API client classes. Instead, it lists the service's versioned client(s) as
+dependencies, and provides convenient factory methods for constructing client
+classes provided by the underlying versioned client libraries. It will choose
+which service version to use by default (although it will generally let you
+override its recommendation if you need to talk to a specific version of the
+service.) For some services, the main client also provides a higher-level
+interface with additional features, convenience methods, or best practices
+built in.
+
+In _most_ cases, we recommend installing the main client gem rather than a
+versioned client gem. This is because the main client will embody the best
+practices for accessing the service, and may also be easier to use. In
+addition, documentation and samples published by Google will generally use the
+main client. However, alternately, if you need to access a specific version of
+a service, and you want to use a lower-level interface, you can bypass the main
+client and instead install a versioned client directly.
+
+Note that some services may not yet have a modern client library (neither a
+main nor a versioned client) available. For these services, there might be a
+_legacy client_ (with a name of the form `google-apis-<service>_<version>`).
+Legacy client libraries have wide coverage across Google services, but may be
+more difficult to use or lack features provided by modern clients.
+<%- end -%>
```

Comment:
This section contains so much information that I think it would benefit from additional structure to make it more approachable. One idea would be to change the header `## Which client should I use?` to `## Client libraries FAQ`, then add a question header above each paragraph. (For example, above the last paragraph, add: **What if I can't find either a main client or a versioned client for the Google service I'm using?**) Another idea is to list the benefits of the main client as bullet points. If you want me to take a shot at restructuring it, happy to provide a draft.

Reply:
@quartzmo Yes, if you have ideas on how this should look, please go ahead and do a draft!

Reply:
Will do.

Reply:
@dazuma I opened dazuma/gapic-generator-ruby#1 with my edits.

Reply:
I reorganized the text around questions as subheadings, similar to your PR, while focusing on the question of why to use main vs versioned clients instead of laying out the entire taxonomy. How does this look?

Reply:
I think what you have now is really good.